src/distributions/slice.rs

```diff
@@ -81,6 +83,13 @@ impl<'a, T> Slice<'a, T> {
             }),
         }
     }
+
+    /// Returns the count of choices in this distribution
+    pub fn choices(&self) -> NonZeroUsize {
```

Comment:
Given the name, I might expect this to return the slice over choices.

Can we not just call this `len`? Or `num_choices`?

Reply:
Sure let's rename it to `num_choices`, gimme a minute